From: Marco Martin <notmart@gmail.com>
Date: Wed, 20 Jul 2016 11:57:08 +0000
Subject: don't try to load layout before kamd starts
X-Git-Url: http://quickgit.kde.org/?p=plasma-workspace.git&a=commitdiff&h=168b987bde98712c8713b839d02c036b0b62d334
---
don't try to load layout before kamd starts

if the status of kamd is anything but running, refuse to
execute load()
this fixes activity creation in default layout initialization
---


--- a/shell/shellcorona.cpp
+++ b/shell/shellcorona.cpp
@@ -302,7 +302,9 @@
 
     connect(m_activityController, &KActivities::Controller::serviceStatusChanged, this, &ShellCorona::load, Qt::UniqueConnection);
 
-    load();
+    if (m_activityController->serviceStatus() == KActivities::Controller::Running) {
+        load();
+    }
 }
 
 QString ShellCorona::shell() const
@@ -330,7 +332,7 @@
 void ShellCorona::load()
 {
     if (m_shell.isEmpty() ||
-        m_activityController->serviceStatus() == KActivities::Controller::Unknown) {
+        m_activityController->serviceStatus() != KActivities::Controller::Running) {
         return;
     }
 

